src/sentry/tasks/performance_detection.py

```diff
@@ -0,0 +1,303 @@
+import hashlib
+import random
+from abc import ABC, abstractmethod
+from datetime import timedelta
+from enum import Enum
+from typing import Any, Dict, List
+
+import sentry_sdk
+
+from sentry import options
+from sentry.eventstore.processing.base import Event
+
+Span = Dict[str, Any]
+TransactionSpans = List[Span]
+PerformanceIssues = Dict[str, Any]
+
+
+class DetectorType(Enum):
+    SLOW_SPAN = "slow_span"
+    DUPLICATE_SPANS = "duplicate_spans"
+    SEQUENTIAL_SLOW_SPANS = "sequential_slow_spans"
+
+
+# Facade in front of performance detection to limit impact of detection on our events ingestion
+def detect_performance_issue(data: Event):
+    try:
+        rate = options.get("store.use-ingest-performance-detection-only")
+        if rate and rate > random.random():
+            # Add an experimental tag to be able to find these spans in production while developing. Should be removed later.
+            sentry_sdk.set_tag("_did_analyze_performance_issue", "true")
+            with sentry_sdk.start_span(
+                op="py.detect_performance_issue", description="none"
+            ) as sdk_span:
+                _detect_performance_issue(data, sdk_span)
+    except BaseException:
```

Comment:
use Exception here. using BaseException catches down `SoftTimeLimitExceeded` from celery and prevents celery from killing the task.

Also logging would be nice here. you can use `sentry_sdk.capture_exception()` or similar

Reply:
Ahh I didn't know about SoftTimeLimit, I originally had it as exception. re: `capture_exception` I don't think it's a good idea to add more load for the exceptions for now, since ideally we roll this out without any impact to ingest.

Reply:
I personally think we can definitely take the load of legitimate bugs, and if there's a massive spike of those it's generally also okay. We just can't (or rather, don't want to) sustain the load for multiple weeks, i.e. if we never fix any bugs.

Reply:
In that case, sounds good to me, I'll add it.